R/repeat_linter.R

```diff
@@ -0,0 +1,46 @@
+#' Repeat linter
+#'
+#' Check that `while` is not used for infinite loops.
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "while (TRUE) { }",
+#'   linters = repeat_linter()
+#' )
+#'
+#'
+#' # okay
+#' lint(
+#'   text = "repeat { }",
+#'   linters = repeat_linter()
+#' )
+#'
+#'
+#' @evalRd rd_tags("repeat_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+repeat_linter <- function() {
+  xpath <- "
+  //WHILE[following-sibling::*[1][self::OP-LEFT-PAREN] and
+  following-sibling::*[2][self::expr[NUM_CONST[text()='TRUE'] and count(*)=1]]
```

Comment:
I would simplify this as following-sibling::expr/NUM_CONST[...]

the node structure of while() loops is pretty simple in terms of what can parse, so no need to be overly defensive IMO.

Reply:
I implemented it as suggested but still had to test that an expression existed at all